### PR TITLE
[CBRD-25176] Failure when creating fk of deduplicate attribute

### PR DESCRIPTION
**Pull request by ctshim**
http://jira.cubrid.org/browse/CBRD-25176

* Failed to create fk index with deduplicate attribute after changing the structure of MIDXKEY

This is to fix a malfunction issue that was affected by changes in the MIDXKEY structure in CBRD-24988.
AS-IS:
     After cloning MIDXKEY, use a trick to compare by reducing the number of columns by only one.
TO-BE:
     Create and use a new key normally.

### CODE REVIEW

File: src/storage/btree.c [hunk 1]
```diff
@@ -6066,6 +6081,73 @@ btree_remake_foreign_key_with_PK (THREAD_ENTRY * thread_p, BTID * btid, DB_VALUE
...
+    }
+  for (i = 0; i < pk_column_cnt; i++)
+    {
+      db_make_null (&(dbvals_ptr[i]));
+    }
+
+  for (i = 0; i < pk_column_cnt; i++)
+    {
+      pr_midxkey_get_element_nocopy (&(fk_key->data.midxkey), i, &dbvals_ptr[i], NULL, NULL);
+    }
```
Comment (hgryoo) [inline src/storage/btree.c]:
```suggestion
  for (i = 0; i < pk_column_cnt; i++)
    {
      db_make_null (&dbvals_ptr[i]);
      pr_midxkey_get_element_nocopy (&(fk_key->data.midxkey), i, &dbvals_ptr[i], NULL, NULL);
    }
```

**Review by hgryoo — APPROVED** [review]

File: src/storage/btree.c (hunk 1 above)
Comment (ctshim) [reply src/storage/btree.c]: Thank you.

File: src/storage/btree.c [hunk 2]
```diff
@@ -5951,6 +5951,7 @@ btree_remake_foreign_key_with_PK (THREAD_ENTRY * thread_p, BTID * btid, DB_VALUE
   DB_MIDXKEY midxkey;
   TP_DOMAIN *tp_dom = NULL;
   TP_DOMAIN *setdomain_ptr = NULL;
+  DB_VALUE new_key_dbvals_ary[8];
```
Comment (beyondykk9) [inline src/storage/btree.c]:
```suggestion
  DB_VALUE new_key_dbval_list[8];
```
"ary" is an unfamiliar postfix.

**Review by beyondykk9 — APPROVED** [review]

**Review by hyunikn — APPROVED** [review]

File: src/storage/btree.c (hunk 2 above)
Comment (ctshim) [reply src/storage/btree.c]: I will modify it with `new_key_dbvals_array`.

**Pull request merged**